reconcile/utils/pagerduty_api.py

```diff
@@ -1,40 +1,78 @@
-import datetime
+from datetime import datetime as dt
+import logging
+from typing import Callable
+from typing import Iterable
+from typing import Optional
+from typing import Protocol
 
-import requests
 import pypd
+import requests
+from pydantic import BaseModel
+from sretoolbox.utils import retry
 
 from reconcile.utils.secret_reader import SecretReader
+from reconcile.utils.secret_reader import SupportsSecret
+
+
+class PagerDutyTargetException(Exception):
+    ...
```

Comment:
nit: should we use `pass` or the ellipsis here? The general style of the code is to use `pass` for this kind of exceptions, but maybe you want to do something else after...